From accb8984ac303c233e6a660296a2bb5308060792 Mon Sep 17 00:00:00 2001
From: Letu Ren <fantasquex@gmail.com>
Date: Mon, 17 Mar 2025 19:00:49 +0800
Subject: [PATCH] Find nanobind first

---
 CMakeLists.txt | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 83bab4701c..5abbef4af1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -780,14 +780,17 @@ endif()
 
 # Both the IREE and MLIR Python bindings require nanobind. We initialize it here
 # at the top level so that everything uses ours consistently.
-if(IREE_BUILD_PYTHON_BINDINGS OR IREE_BUILD_COMPILER)
-  include(FetchContent)
-  FetchContent_Declare(
-    nanobind
-    GIT_REPOSITORY https://github.com/wjakob/nanobind.git
-    GIT_TAG        0f9ce749b257fdfe701edb3cf6f7027ba029434a # v2.4.0
-  )
-  FetchContent_MakeAvailable(nanobind)
+if(IREE_BUILD_PYTHON_BINDINGS)
+  find_package(nanobind)
+  if (NOT nanobind_FOUND)
+    include(FetchContent)
+    FetchContent_Declare(
+      nanobind
+      GIT_REPOSITORY https://github.com/wjakob/nanobind.git
+      GIT_TAG        0f9ce749b257fdfe701edb3cf6f7027ba029434a # v2.4.0
+    )
+    FetchContent_MakeAvailable(nanobind)
+  endif()
 endif()
 
 if(NOT IREE_BUILD_COMPILER)
-- 
2.48.1

