From 09a503dc94382798a1ee2f23b4b6894a12bc8207 Mon Sep 17 00:00:00 2001
From: SharzyL <me@sharzy.in>
Date: Mon, 26 Aug 2024 20:08:47 +0800
Subject: [PATCH 2/3] disable NaN normalization

---
 riscv/v_ext_macros.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/riscv/v_ext_macros.h b/riscv/v_ext_macros.h
index e5ea1045..202fde9c 100644
--- a/riscv/v_ext_macros.h
+++ b/riscv/v_ext_macros.h
@@ -1518,7 +1518,7 @@ reg_t index[P.VU.vlmax]; \
                 softfloat_exceptionFlags |= softfloat_flag_invalid; \
                 set_fp_exceptions; \
               } \
-              P.VU.elt<uint16_t>(rd_num, 0, true) = defaultNaNF16UI; \
+              P.VU.elt<uint16_t>(rd_num, 0, true) = vd_0.v; \
             } else { \
               P.VU.elt<uint16_t>(rd_num, 0, true) = vd_0.v; \
             } \
@@ -1531,7 +1531,7 @@ reg_t index[P.VU.vlmax]; \
                 softfloat_exceptionFlags |= softfloat_flag_invalid; \
                 set_fp_exceptions; \
               } \
-              P.VU.elt<uint32_t>(rd_num, 0, true) = defaultNaNF32UI; \
+              P.VU.elt<uint32_t>(rd_num, 0, true) = vd_0.v; \
             } else { \
               P.VU.elt<uint32_t>(rd_num, 0, true) = vd_0.v; \
             } \
@@ -1544,7 +1544,7 @@ reg_t index[P.VU.vlmax]; \
                 softfloat_exceptionFlags |= softfloat_flag_invalid; \
                 set_fp_exceptions; \
               } \
-              P.VU.elt<uint64_t>(rd_num, 0, true) = defaultNaNF64UI; \
+              P.VU.elt<uint64_t>(rd_num, 0, true) = vd_0.v; \
             } else { \
               P.VU.elt<uint64_t>(rd_num, 0, true) = vd_0.v; \
             } \
-- 
2.43.0

