From 0a8636ceecac4f0b078b86fd10d54dfb7fa531bf Mon Sep 17 00:00:00 2001
From: Shupei Fan <dymarkfan@outlook.com>
Date: Mon, 18 Aug 2025 02:45:26 +0000
Subject: [PATCH 2/3] [t1-fix] relax vlen limit

---
 disasm/isa_parser.cc | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/disasm/isa_parser.cc b/disasm/isa_parser.cc
index e37b72f3..1f092389 100644
--- a/disasm/isa_parser.cc
+++ b/disasm/isa_parser.cc
@@ -458,10 +458,6 @@ isa_parser_t::isa_parser_t(const char* str, const char *priv)
   }
 #endif
 
-  if (vlen > 4096) {
-    bad_isa_string(str, "Spike does not currently support VLEN > 4096b");
-  }
-
   if ((vlen != 0) ^ (elen != 0)) {
     bad_isa_string(str, "Invalid Zvl/Zve configuration");
   }
-- 
2.43.5

